huawei_lte_api/api/User.py

```diff
@@ -73,10 +74,20 @@ def _login(self, password_type: PasswordTypeEnum=PasswordTypeEnum.BASE_64) -> bo
         return result == ResponseEnum.OK.value
 
     def login(self, force_new_login: bool=False) -> bool:
-        try:
-            state_login = self.state_login()
-        except ResponseErrorNotSupportedException:
-            return True
+        retries = 5
+        for i in range(retries):
+            try:
+                state_login = self.state_login()
+            except ConnectionError as e:
+                # Some models reportedly close the connection if we attempt to access login state too soon after
+                # setting up the session etc. In that case, retry a few times. The error is reported to be
+                # ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
+                if i == retries - 1 or not isinstance(
+                        getattr(e, '__context__', getattr(e, '__cause__', None)), ConnectionResetError):
```

Comment:
This second condition reportedly doesn't work: https://github.com/Salamek/huawei-lte-api/issues/65#issuecomment-589870278

The intent here is to be more specific what we want to trigger the retries on. Maybe there's a fault in the code that I cannot see, or the context/cause is something else than a `ConnectionResetError` or its subclass (`RemoteDisconnected` per docs is)).

Anyway the 2nd check can be dropped altogether, or something like '"connection aborted" not in str(e).lower()` used, whichever approach @Salamek prefers.

Reply:
Dropped it below, hope this is good to go now.